Testing

amita-roy opened the pull request:
- I'm working on the testing branch to add tests for the math-magicians website.
- All the tests are in the src/test directory.
- Test has been added via jest and a snapshot for all three routes has also been added.
- For randomly generated quotes,  I have used jest mock for quotes route.

## In the milestone1 pull request  I have declared that I was having an issue with stylelint so I have to make changes and mentioned that what changes I'm making, after that my other milestones also got approved with that change.

amita-roy committed:
Add test for calculate.js function

amita-roy committed:
Add test for homepage, calculator page

amita-roy committed:
Add snapshots

amita-roy committed:
Add snapshot testing

amita-roy committed:
Update README

krys2fa reviewed (changes requested):
# Status: Marked as Invalid! ⚠️  
Please note that your Code Review Request is being marked as invalid. 
This is because the configs in .github/workflows/ must be exactly the same as that in the [Microverse linter config repo](https://github.com/microverseinc/linters-config/blob/master/react-redux/.github/workflows/linters.yml). 

💡 Do not make any changes in config files - they represent style guidelines that you share with your team - which is a group of all Microverse students.
If you think that the change is necessary - open a [Pull Request in this repository](https://github.com/microverseinc/linters-config/blob/master/README.md#contributing).


This Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready!

Amita-Roy commented:
### In the milestone1 pull request, I have declared that I was having an issue with stylelint so I have to make changes and mentioned that what changes I'm making, after that my other milestones also got approved with that change.

Torres-ssf commented on src/test/calculate.test.js:
```diff
@@ -0,0 +1,49 @@
+import Calculate from '../logic/calculate';
```
- [x] It seems the modifier operator `%` is not being tested. We need to implement at least a test for it.

Torres-ssf commented on src/test/calculatorpage.test.js:
```diff
@@ -0,0 +1,33 @@
+  test('has element with className calculator', () => {
+    render(<Calculator />);
+    const { container } = render(<div className="calculator" />);
+    // const element = screen.getByRole('display');
```
- [x] Let's remove this commented code.

Torres-ssf reviewed (changes requested):
### Changes required :muscle: 
Hi @Amita-Roy :hand: 
Awesome job on testing your application so far.
But we need to make so tests before moving on.

- [x] We have one test failing, we need to fix it.

Happy coding :computer: 
**Sérgio Torres**

amita-roy committed:
Add test for % operator and remove the comment out code

Amita-Roy commented:
- Added a test for % operator
- Removed the commented out code
- All tests are passing, please don’t refer to the below screenshot

<img width="1316" alt="Screenshot 2021-03-28 at 11 59 52 AM" src="https://user-images.githubusercontent.com/47393794/112748651-6faafd80-8fbd-11eb-8c91-60f820fee4c3.png">

Torres-ssf reviewed (approved):
### PR Approved :clap: :tada: 
Hello again @Amita-Roy :hand: 
Great, all required changes were implemented.
By the way, your calculator style is awesome, it's looking really nice.
Good luck with the next project

Happy coding :computer: 
**Sérgio Torres**
